rails/db/migrate/20220121213539_update_portal_clazz_is_archived.rb

```diff
@@ -0,0 +1,17 @@
+class UpdatePortalClazzIsArchived < ActiveRecord::Migration[6.1]
+  module Portal
+  end
+
+  class Portal::Clazz < ApplicationRecord
+    self.table_name = :portal_clazzes
+  end
+  class Portal::TeacherClazz < ApplicationRecord
+    self.table_name = :portal_teacher_clazzes
+  end
+
+  def change
+    Portal::TeacherClazz.find_each(batch_size: 100) do |tc|
+      Portal::Clazz.where(id: tc.clazz_id).update_all(is_archived: !tc.active)
```

Comment:
This update_all is a little weird. There should only be one Portal::Class with a particular id.  But I think this should work fine.

Reply:
Ah, right.